### PR TITLE
Use bounded Ipopt_jll from AmplNLWriter

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -63,6 +63,7 @@ ForwardDiff = "0.10.3"
 FunctionWrappersWrappers = "0.1"
 Graphs = "1.5.2"
 IfElse = "0.1"
+Opopt_jll = "=300.1400.400"
```
Comment (Vaibhavdixit02) [inline Project.toml:66]:
```suggestion
Ipopt_jll = "=300.1400.400"
```

File: Project.toml [hunk 2]
```diff
@@ -63,6 +63,7 @@ ForwardDiff = "0.10.3"
 FunctionWrappersWrappers = "0.1"
 Graphs = "1.5.2"
 IfElse = "0.1"
+Ipopt_jll = "=300.1400.400"
```
Comment (ChrisRackauckas) [inline Project.toml:66]: Why is this bound here in this repo?

Comment (Vaibhavdixit02) [reply Project.toml:66]: Because it is a test dependency here as well https://github.com/SciML/ModelingToolkit.jl/blob/master/Project.toml#L93

Comment (YingboMa) [reply Project.toml:66]: Can we just not use `Ipopt_jll` in the test?

Comment (Vaibhavdixit02) [reply Project.toml:66]: https://github.com/SciML/Optimization.jl/commit/88fc980a385c795cb0b7456e46151bdef9ecbafe I thought so too, but the tests were still failing here
https://github.com/SciML/ModelingToolkit.jl/actions/runs/4439811295/jobs/7852885734#step:6:70
https://github.com/SciML/ModelingToolkit.jl/actions/runs/4439811295/jobs/7852885734#step:6:92

Comment (ChrisRackauckas) [reply Project.toml:66]: Is it using the right version?

Comment (Vaibhavdixit02) [reply Project.toml:66]: https://github.com/SciML/ModelingToolkit.jl/actions/runs/4472354692/jobs/7858412913#step:6:331 no it's not

**Pull request closed without merging**